model/pdata/generated_metrics.go

```diff
@@ -1366,6 +1366,26 @@ func (ms HistogramDataPoint) SetFlags(v MetricDataPointFlags) {
 	(*ms.orig).Flags = uint32(v)
 }
 
+// Min returns the min associated with this HistogramDataPoint.
+func (ms HistogramDataPoint) Min() OptionalDouble {
+	return OptionalDouble{orig: ((*ms.orig).Min)}
+}
+
+// SetMin replaces the min associated with this HistogramDataPoint.
+func (ms HistogramDataPoint) SetMin(v OptionalDouble) {
+	(*ms.orig).Min = v.orig
+}
```

Comment:
Another approach:

```golang
// Min returns the min associated with this HistogramDataPoint.
func (ms HistogramDataPoint) Min() float64 {
	return (*ms.orig).Min.DoubleValue
}

func (ms HistogramDataPoint) HasMin() bool {
	!(*ms.orig).Min.IsEmpty()
}


func (ms HistogramDataPoint) SetMin(v float64) {
	(*ms.orig).Min.DoubleValue = v
}
```

Reply:
Are you suggesting we do away with the `OptionalDouble` type in pdata? I don't have a strong preference to keeping it and this approach would give us essentially the same functionality without an additional type.

Reply:
That is my idea.. What do you think?

Reply:
Just tried it and I agree it's better. I've added the `HasMin` `HasMax` methods to metrics.go instead of creating a new optional field generator. We can always add one later if needed.

Reply:
Decided to add a template for optional fields after all, since this will be re-used for min/max and sum in the near future. PTAL @bogdandrutu